src/sagas/identity.js

```diff
@@ -170,14 +187,37 @@ function * handleAddNewCategory(action) {
       name: action.payload.value,
       desc: '',
       identity: selectedIdentityId,
-      claims: [],
     },
   };
 
   yield put(setNewCategory(newCategory[newCategoryId]));
   yield call(updateClaimCategoryStorage);
 }
 
+function * handleMoveClaims(action) {
+  const { targetCategory } = action.payload;
+  const selectedClaims = yield select(selectSelectedClaims);
+  yield all(selectedClaims.toJS().map((claim) => put(updateCategoryForClaim(claim.id, targetCategory.get('id')))));
```

Comment:
No need to dispatch a series of actions.
Create a custom action that will take a list of claims and change their category.
Having that kind of action would make the case where you only need to update one claim, simple as sending a list of one element.